aggregator_core/src/datastore/models.rs

```diff
@@ -613,6 +614,18 @@ impl<T> Lease<T> {
     pub fn lease_attempts(&self) -> usize {
         self.lease_attempts
     }
+
+    /// Returns how long remains until the expiry time, with an optional clock skew allowance.
+    /// The math saturates, because  we want to timeout immediately if any of these
+    /// subtractions would underflow.
+    pub fn remaining_lease_duration(&self, current_time: &Time, skew_seconds: u64) -> StdDuration {
```

Comment:
Saturating math and depending on underflow for control logic is fine, but it always gives me pause. I think it'd be nice to write a test on this method both to make sure some idiot like me doesn't introduce regressions but also to express and document our intended semantics for leases, especially in edge cases like where `self.lease_expiry_time` is 0.